### ISSUE TITLE
Fix digital payoff touch logic

### ISSUE THREAD

**Issue by FedererKK** [vyper-protocol/vyper-core#80]
Right now 'CALL' is using => whereas 'PUT' is only using <, leaving the = out might alter the payoff?

https://github.com/vyper-protocol/vyper-core/blob/daa6fe99f8c393339b6416e6af4187ce6df969a6/programs/redeem-logic-digital/src/lib.rs#L150

Also this might be one of those cases where we use the Pyth confidence interval eg, if the settlement price is in the confidence interval the payout is halved. I believe Zeta does something similar too.

**Comment by thegreat-vanderlinde** [comment]
> Right now 'CALL' is using => whereas 'PUT' is only using <, leaving the = out might alter the payoff?

usually having the `equal` only on "one side of the interval" is common in sw dev, but in this case we have the `is_call` flag that selects the condition, probably we can include the equal on both.

```rust
 if (is_call && new_spot >= strike) || (!is_call && new_spot <= strike) { 
```

@vantessel what do you think?

tldr;
`strike` and `new_spot` are both decimals, so the probability of having the same value is quite low, but for the sake of correctness that's definitely a good point.

> Also this might be one of those cases where we use the Pyth confidence interval eg, if the settlement price is in the confidence interval the payout is halved. I believe Zeta does something similar too.

This instead could be something harder to accomplish: having the plugin interfaces to observe all the `redeem logic plugin`s needs to behavior the same way indipently to the `rate plugin` sibling (switchboard / pyth).

**Comment by vantessel** [comment]
This is a common convention for digitals, as digital call + put = guaranteed payment at maturity

https://github.com/vyper-protocol/vyper-core/blob/daa6fe99f8c393339b6416e6af4187ce6df969a6/programs/redeem-logic-digital/src/lib.rs#L6
